static/js/message_events.js

```diff
@@ -74,7 +74,14 @@ exports.insert_new_messages = function insert_new_messages(messages, sent_by_thi
         message_util.add_new_messages(messages, home_msg_list);
 
         if (narrow_state.filter().can_apply_locally()) {
-            render_info = message_util.add_new_messages(messages, message_list.narrowed);
+            let opts = {};
+            if (!current_msg_list.data.fetch_status.has_found_newest()) {
+                // We can't render new messages if we haven't found the latest messages
+                // in the view without making it appear the message was just after the
+                // previously rendered messages when that's inaccurate.
+                opts = {defer_rendering: true};
+            }
+            render_info = message_util.add_new_messages(messages, message_list.narrowed, opts);
```

Comment:
I think we need the behavior to change for the `message_util.add_new_messages` in the else clause just a bit below, so this is probably the wrong place to do our conditional; I think we can just not modify this file and instead change `message_util.add_new_messages` (or lower down the call stack).  I'll post a common there on how I think this should work.

Reply:
Yes I can confirm this is required for home view too. I tested this by setting the `found_newest` to False every time a message is received.